packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,239 @@
+            );
+        }
+        return null;
+    }
+
+    public renderUnselectedGlyph(): React.ReactNode {
+        if (typeof this.props.unselectedGlyph === "function") {
+            return this.props.unselectedGlyph(
+                get(this.props, "managedClasses.actionToggle_unselectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.actionToggle") || "";
+
+        if (this.props.disabled) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__disabled"
+            )}`;
+        }
+
+        if (this.state.selected) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__selected"
+            )}`;
+        }
+
+        if (this.isSingleElement()) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__singleElement"
+            )}`;
+        }
+
+        switch (this.props.appearance) {
+            case ActionToggleAppearance.primary:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__primary"
+                )}`;
+                break;
+            case ActionToggleAppearance.lightweight:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__lightweight"
+                )}`;
+                break;
+            case ActionToggleAppearance.justified:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__justified"
+                )}`;
+                break;
+            case ActionToggleAppearance.outline:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.actionToggle__outline"
+                )}`;
+                break;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     * Checks to see if the toggle is displaying both glyph and text or not
+     */
+    private isSingleElement(): boolean {
+        if (this.state.selected) {
+            if (
+                isNullOrWhiteSpace(this.props.selectedText) ||
+                isNullOrUndefined(this.props.selectedGlyph)
+            ) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            if (
+                isNullOrWhiteSpace(this.props.unselectedText) ||
+                isNullOrUndefined(this.props.unselectedGlyph)
+            ) {
+                return true;
+            } else {
+                return false;
+            }
+        }
```

Comment:
Can you provide some insight on the usefulness of this? Looking to understand as it's not clear to me.

Reply:
We need to add a margin between glyph and content when both are present, but not when there is only one as stuff won't be centered.  This function evaluates that so we can add the right classes to add that margin.